backend/experiment/rules/matching_pairs_2025.py

```diff
@@ -50,26 +61,202 @@ def next_round(self, session):
             trial = self.get_matching_pairs_trial(session)
             actions.append(trial)
             return actions
-        else:
-            feedback_info = self.feedback_info()
-            score = Final(
-                session,
-                title="Score",
-                final_text=self.final_text(session),
-                button={"text": "Play again", "link": self.get_play_again_url(session)},
-                rank=self.rank(session, exclude_unfinished=False),
-                feedback_info=feedback_info,
-                percentile=session.percentile_rank(exclude_unfinished=False),
+
+        # Finish session and show final view
+        session.finish()
+        session.save()
+
+        return self._get_final_actions(session)
+
+    def _get_final_actions(self, session: Session):
+        accumulated_score = session.participant.session_set.aggregate(total_score=models.Sum("final_score"))[
+            "total_score"
+        ]
+        score = Final(
+            session,
+            title="Score",
+            total_score=accumulated_score,
+            final_text=self._final_text(session),
+            button={"text": "Next game", "link": self.get_experiment_url(session)},
+            rank=self.rank(session, exclude_unfinished=False),
+            percentile=session.percentile_rank(exclude_unfinished=False),
+        )
+
+        return [score]
+
+    def get_matching_pairs_trial(self, session: Session):
+        player_sections = self._select_sections(session)
+        random.seed(self.random_seed)
```

Comment:
My previous question was aiming at why you would set a ramdom seed here before shuffling. That was used for enforcing a fixed order, but I don't think it's needed here?